libtorrent.cpython-*.so fails to load libtorrent-rasterbar.so.10.0.0.
Give it the proper SHARED_LIB.

Add include path to find boost headers.

commit 1ec11e4e9b2d12d26f4661a14743822c91d76003
from: arvidn <arvid@libtorrent.org>
date: Fri Jul 22 22:12:42 2022 UTC
via: Arvid Norberg <arvid.norberg@gmail.com>

back-port Jamfile version fix from RC_2_0

Index: Jamfile
--- Jamfile.orig
+++ Jamfile
@@ -18,6 +18,8 @@ ECHO "CXXFLAGS =" $(CXXFLAGS) ;
 ECHO "LDFLAGS =" $(LDFLAGS) ;
 ECHO "OS =" [ os.name ] ;
 
+jam-version = [ modules.peek : JAM_VERSION ] ;
+
 if $(BOOST_ROOT)
 {
 	ECHO "building boost from source directory: " $(BOOST_ROOT) ;
@@ -46,7 +48,7 @@ else
 
 # we need version numbers in the form X.Y.Z in order to trigger the built-in
 # support for generating symlinks to the installed library
-VERSION = 10.0.0 ;
+VERSION = ${LIBtorrent-rasterbar_VERSION} ;
 
 rule linking ( properties * )
 {
@@ -756,6 +758,7 @@ ED25519_SOURCES =
 local usage-requirements =
 	<include>./include
 	<include>./include/libtorrent
+	<include>${LOCALBASE}/include
 	<include>/usr/sfw/include
 	<variant>release:<define>NDEBUG
 	<define>_FILE_OFFSET_BITS=64
@@ -823,8 +826,10 @@ rule install-paths ( properties * )
 	# package.paths was introduced in boost-1.70 (2018.02)
 	# however, boost build's versioning scheme changed in boost-1.71 to version
 	# 4.0
-	local boost-build-version = [ SPLIT_BY_CHARACTERS [ version.boost-build ] : "-" ] ;
-	if [ version.version-less [ SPLIT_BY_CHARACTERS $(boost-build-version[1]) : "." ] : 2018 03 ]
+	# so, if versions are 4.0+ we want to use package.paths, but if it's a year,
+	# say 2018, that means it's old and we use the fallback below. Any version <
+	# 1990 is considered the 4.0 and later numbering scheme.
+	if [ version.version-less 1990 0 : $(jam-version) ]
 	{
 		import option ;
 		import property ;
